src/constructs/autoscaling/asg.test.ts

```diff
@@ -301,4 +303,120 @@ describe("The GuAutoScalingGroup", () => {
+
+    new GuAutoScalingGroup(stack, "AutoscalingGroup", {
+      app: "TestApp",
+      userData: "SomeUserData",
+      vpc,
+    });
+
+    expect(stack).toHaveResource("AWS::IAM::Role", {
+      AssumeRolePolicyDocument: {
+        Version: "2012-10-17",
+        Statement: [
+          {
+            Action: "sts:AssumeRole",
+            Effect: "Allow",
+            Principal: { Service: { "Fn::Join": ["", ["ec2.", { Ref: "AWS::URLSuffix" }]] } },
+          },
+        ],
+      },
+      Tags: alphabeticalTags([
+        { Key: "App", Value: "TestApp" },
+        { Key: "gu:cdk:version", Value: "TEST" },
+        { Key: "Stack", Value: "test-stack" },
+        { Key: "Stage", Value: { Ref: "Stage" } },
+      ]),
+    });
+  });
+
+  test("passing in an instance role overrides the default", () => {
+    const stack = simpleGuStackForTesting();
+
+    new GuAutoScalingGroup(stack, "AutoscalingGroup", {
+      app: "TestApp",
+      userData: "UserData",
+      vpc,
+      role: new GuInstanceRole(stack, {
+        app: "TestApp",
+        additionalPolicies: [
+          new GuAllowPolicy(stack, "SomePolicy", {
+            actions: ["some:Action"],
+            resources: ["some:Resource"],
+          }),
+        ],
+      }),
+    });
+
+    expect(stack).toHaveResource("AWS::IAM::Role", {
+      AssumeRolePolicyDocument: {
+        Statement: [
+          {
+            Action: "sts:AssumeRole",
+            Effect: "Allow",
+            Principal: { Service: { "Fn::Join": ["", ["ec2.", { Ref: "AWS::URLSuffix" }]] } },
+          },
+        ],
+        Version: "2012-10-17",
+      },
+    });
+
+    expect(stack).toHaveResource("AWS::IAM::Policy", {
+      PolicyDocument: {
+        Version: "2012-10-17",
+        Statement: [
+          {
+            Action: "some:Action",
+            Effect: "Allow",
+            Resource: "some:Resource",
+          },
+        ],
+      },
+    });
+  });
+
+  test("has sensible default scaling capacities per stage based on minimum capacity", () => {
+    const stack = simpleGuStackForTesting();
+
+    new GuAutoScalingGroup(stack, "AutoscalingGroup", {
+      app: "TestApp",
+      userData: "SomeUserData",
+      vpc,
+    });
+
+    const json = SynthUtils.toCloudFormation(stack) as SynthedStack;
+
+    expect(json.Mappings).toEqual({
+      stagemapping: {
+        CODE: { minInstances: 1, maxInstances: 2 },
+        PROD: { minInstances: 3, maxInstances: 6 },
+      },
+    });
+  });
+
+  test("scaling capacities can be overriden", () => {
+    const stack = simpleGuStackForTesting();
+
+    new GuAutoScalingGroup(stack, "AutoscalingGroup", {
+      app: "TestApp",
+      userData: "SomeUserData",
+      vpc,
+      stageDependentProps: {
+        [Stage.CODE]: { minimumInstances: 2 },
+        [Stage.PROD]: { minimumInstances: 5 },
+      },
+    });
+
+    const json = SynthUtils.toCloudFormation(stack) as SynthedStack;
+
+    expect(json.Mappings).toEqual({
```

Comment:
For completeness, both of these last two tests should probably also check that we are referencing these mappings correctly i.e. 

```typescript
expect(stack).toHaveResource("AWS::AutoScaling::AutoScalingGroup", {
  MinSize: {
    "Fn::FindInMap": [
      "stagemapping",
       {
          Ref: "Stage",
        },
        "minInstances",
      ],
    },
    MaxSize: {
      "Fn::FindInMap": [
        "stagemapping",
          {
          Ref: "Stage",
          },
          "maxInstances",
      ],
    },
});
```

If we add this here, then we can probably remove an existing test (`Uses Find In Map correctly to reference capacity mappings`) too.

Reply:
Good shout, and done!